packages/event-handler/tests/unit/rest/converters.test.ts

```diff
@@ -396,6 +396,18 @@ describe('Converters', () => {
       expect(result.statusCode).toBe(204);
       expect(result.body).toBe('');
     });
+
+    it('handles compressed response body', async () => {
+      const response = new Response('Hello World', {
+        status: 200,
+        headers: {
+          'content-encoding': 'gzip',
+        },
+      });
+
+      const result = await webResponseToProxyResult(response);
+      expect(result.isBase64Encoded).toBe(true);
```

Comment:
We need to check the body here too make sure that the logic that converts it to a base64 string works.